packages/react-native-executorch/common/rnexecutorch/threads/GlobalThreadPool.h

```diff
@@ -0,0 +1,77 @@
+// GlobalThreadPool.h
+#pragma once
+
+#include <executorch/extension/threadpool/cpuinfo_utils.h>
+#include <memory>
+#include <mutex>
+#include <rnexecutorch/Log.h>
+#include <rnexecutorch/threads/HighPerformanceThreadPool.h>
+
+namespace rnexecutorch {
+namespace threads {
+
+class GlobalThreadPool {
+private:
+  inline static std::unique_ptr<HighPerformanceThreadPool> instance;
+  inline static std::once_flag initFlag;
+
+  GlobalThreadPool() = delete;
```

Comment:
Automatically generated ctor c-assignment are deleted implicitly because `instance` does not support copying. But moving ctor and assignment will be generated. You probably want to delete them as well ;)